### PR TITLE
refactor: small optimize for sql query

### PR DESCRIPTION
**Pull request by appleboy**
* Remove unused function `RepositoriesWithUsers`.
* SQL improvement for `repo.GetOwner` loop condition.

cc @lunny

### CODE REVIEW

**Comment by bkcsoft** [comment]
If it compiles and runs you have my LGTM :)

File: models/repo.go [hunk 1]
```diff
@@ -1184,26 +1184,22 @@ func CountUserRepositories(userID int64, private bool) int64 {
...
+		OrderBy(opts.OrderBy).
+		Find(&repos); err != nil {
+		return nil, fmt.Errorf("Repo: %v", err)
 	}
 
-	for i := range repos {
-		if err = repos[i].GetOwner(); err != nil {
-			return nil, err
-		}
+	if err = repos.loadAttributes(x); err != nil {
```
Comment (lunny) [inline models/repo.go]: But maybe sometime, we don't need to `loadAttributes`?

File: models/repo.go [hunk 2]
```diff
@@ -1848,10 +1852,8 @@ func SearchRepositoryByName(opts *SearchRepoOptions) (repos RepositoryList, _ in
 		return nil, 0, fmt.Errorf("Repo: %v", err)
 	}
 
-	if opts.Starred {
-		if err = repos.loadAttributes(x); err != nil {
-			return nil, 0, fmt.Errorf("LoadAttributes: %v", err)
-		}
+	if err = repos.loadAttributes(x); err != nil {
```
Comment (lunny) [inline models/repo.go]: Why always loadAttributes?

**Review by lunny — CHANGES_REQUESTED** [review]

File: models/repo.go (hunk 1 above)
Comment (appleboy) [reply models/repo.go]: Yes you're right. Ony two condition need to `loadAttributes`.

1. `Searcher` is not nil.
2. Star page of user profile

https://github.com/go-gitea/gitea/pull/940/commits/974dea2b36c59cf74dd2fc055e302a5d1aa2d785

File: models/repo.go (hunk 2 above)
Comment (appleboy) [reply models/repo.go]: Done! https://github.com/go-gitea/gitea/pull/940/commits/974dea2b36c59cf74dd2fc055e302a5d1aa2d785

File: models/repo.go (hunk 1 above)
Comment (lunny) [reply models/repo.go]: I prefer add a new field `IfLoadAttributes bool` for `SearchRepoOptions`. Then we can not hard code there.

Comment (appleboy) [reply models/repo.go]: I forgot one place need to update.

See new commit: https://github.com/go-gitea/gitea/pull/940/commits/aa9fac15fc3d44e5aa221e8681e364f5ac1658b1

We don't need a new field `IfLoadAttributes`, just two condition need `GetOwner` data:

`opts.Searcher != nil || opts.Starred`

Comment (lunny) [reply models/repo.go]: OK.

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
LGTM

**Pull request merged**